src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+                 BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+
+            int supplyTotal = 0;
+            int buyTotal = 0;
+            String line;
+
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (parts.length == 2) {
+                    String operationType = parts[0];
+                    int amount = Integer.parseInt(parts[1]);
+                    if ("supply".equals(operationType)) {
+                        supplyTotal += amount;
+                    } else if ("buy".equals(operationType)) {
+                        buyTotal += amount;
+                    }
+                }
+            }
+
+            int result = supplyTotal - buyTotal;
+
+            writer.write("supply," + supplyTotal);
+            writer.newLine();
+            writer.write("buy," + buyTotal);
+            writer.newLine();
+            writer.write("result," + result);
 
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
     }
```

Comment:
```suggestion
    public void getStatistic(String fromFileName, String toFileName) {
        String dataFromFile = readFile(fromFileName);
        String report = createReport(dataFromFile);
        writeToFile(report, toFileName);
    }
```
![image](https://github.com/mate-academy/jv-work-with-file/assets/20326858/5c9eacb4-c746-4223-a083-34dca85b580c)